spacecmd/src/spacecmd/errata.py

```diff
@@ -92,31 +92,34 @@ def complete_errata_apply(self, text, line, beg, end):
     return self.tab_complete_errata(text)
 
 
-def do_errata_apply(self, args, only_systems=None):
+def do_errata_apply(self, args, errata=None, only_systems=None):
     arg_parser = get_argument_parser()
+    arg_parser.add_argument('errata', nargs="*")
     arg_parser.add_argument('-s', '--start-time')
 
     args, options = parse_command_arguments(args, arg_parser)
     only_systems = only_systems or []
+    errata = options.errata or errata
+    del options.errata
 
-    if not args:
+    if not args and not errata:
         self.help_errata_apply()
         return 1
 
     # get the start time option
     # skip the prompt if we are running with --yes
     # use "now" if no start time was given
     if is_interactive(options) and not self.options.yes:
-        options.start_time = prompt_user(_('Start Time [now]:'))
-        options.start_time = parse_time_input(options.start_time)
+        start_time = prompt_user(_('Start Time [now]:'))
+        start_time = parse_time_input(start_time)
     else:
         if not options.start_time:
-            options.start_time = parse_time_input('now')
+            start_time = parse_time_input('now')
         else:
-            options.start_time = parse_time_input(options.start_time)
+            start_time = parse_time_input(options.start_time)
 
     # allow globbing and searching via arguments
-    errata_list = self.expand_errata(args)
+    errata_list = self.expand_errata(errata)
```

Comment:
Isn't this redundant if `do_errata_apply` is called from `do_system_applyerrata`?

Reply:
Yes, it's redundant. Good catch :+1: